recipes/st_tree/all/conanfile.py

```diff
@@ -1,42 +1,58 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy, rmdir
+from conan.tools.layout import basic_layout
+from conan.tools.cmake import CMake, CMakeToolchain
 import os
 
-from conans.errors import ConanInvalidConfiguration
-from conans import ConanFile, CMake, tools
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.52.0"
 
 class STTreeConan(ConanFile):
     name = "st_tree"
+    description = "A fast and flexible c++ template class for tree data structures"
     license = "Apache-2.0"
     url = "https://github.com/conan-io/conan-center-index"
-    description = "A fast and flexible c++ template class for tree data structures"
-    topics = ("stl", "container", "data-structures")
     homepage = "https://github.com/erikerlandson/st_tree"
+    topics = ("stl", "container", "data-structures", "tree", "header-only")
+    package_type = "header-library"
+    settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 11
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def _configure_cmake(self):
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["ENABLE_TESTS"] = False
+        tc.generate()
+
+    def build(self):
         cmake = CMake(self)
-        cmake.configure(source_folder=self._source_subfolder)
-        return cmake
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("LICENSE", "licenses", self._source_subfolder)
-
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
 
-        tools.rmdir(os.path.join(self.package_folder, "lib"))
-
-    def package_id(self):
-        self.info.header_only()
+        rmdir(self, os.path.join(self.package_folder, "lib"))
 
     def package_info(self):
-        self.cpp_info.filenames["cmake_find_package"] = "st_tree"
```

Comment:
Is this not needed anymore? Will they have proper defaults when removed?

I am just curious, the [Migrating legacy cpp_info attributes to set_property()](https://docs.conan.io/en/1.53/migrating_to_2.0/properties.html) does mention how to migrate these but does not mention removing is a valid migration option for Conan 2
(PS: I am looking at different PRs to understand how to migrate our packages once we start doing it)

Reply:
Hi @CJCombrink,
We are leaving those for some time to not break compatibility with v1 users that use the legacy generators, but it will be removed from all recipes soon. If you are using v1 with CMakeToolchain, CMakeDeps, and the other generators compatible with v2 you can remove those from your recipes.
Also note that adding test_v1_package for new recipes is not mandatory any more.